Rakefile

```diff
@@ -1,42 +1,76 @@
-%w{ bundler/setup rubygems fileutils uri net/https tmpdir digest/md5 ./doc/markit }.each do |file|
+%w{ bundler/setup yaml uri net/https tmpdir digest/md5 ./doc/markit }.each do |file|
   require file
 end
 
 # Immediately sync all stdout so that it's immediately visible, e.g. on appveyor
 $stdout.sync = true
 $stderr.sync = true
 
-VERSION = '3.1-SNAPSHOT'
-BASE_DIR = File.expand_path('.', File.dirname(__FILE__))
-TARGET_DIR  = "#{BASE_DIR}/target"
-BUILD_DIR   = "#{BASE_DIR}/target/build"
-CACHE_DIR   = "#{BASE_DIR}/target/cache"
-ZIP_EXE = 'C:\Program Files\7-Zip\7z.exe'
+module EnvironmentOptions
+  BASE_DIR = File.expand_path('.', File.dirname(__FILE__))
+  VERSION = '3.1-SNAPSHOT'
+  def version
+    VERSION
+  end
+  def base_dir
+    return BASE_DIR
+  end
+  def target_dir
+    ddir=ENV["TARGET_DIR"]
+    ddir||=File.join(base_dir,"target")
+    File.expand_path(ddir)
```

Comment:
how about `File.expand_path(ENV["TARGET_DIR"] || File.join(base_dir, "target")` instead?

for my own taste a bit more readable as a one-liner, but not being picky here